### DISCUSSION TITLE
看了emp相关wiki，有一点我不是太明白，还希望解答一下。应用之间远程调用的意义是？

### DISCUSSION THREAD

**Discussion by tyawei** [empjs/emp#203] [General]
wiki上讲述了 vue 应用调用react（或者反过来），不过这样做的意义，只是调用react应用中的通用组件吗？
我目前怀疑这功能的意义：
1. 如果组件远程调用只是为了通用，为何不提取发布 npm？
2. 如果不是为了通用，那么应用之间大可能具有业务相关性，为什么又要应用隔离，同时这貌似和微前端理念不符合。
3. 如果我的系统暂时不存在react应用，目前只有 vue应用、jquery老系统，远程调用怎么做呢？

**Comment by ron0115** [comment]
> wiki上讲述了 vue 应用调用react（或者反过来），不过这样做的意义，只是调用react应用中的通用组件吗？
> 我目前怀疑这功能的意义：
> 
> 1. 如果组件远程调用只是为了通用，为何不提取发布 npm？
> 2. 如果不是为了通用，那么应用之间大可能具有业务相关性，为什么又要应用隔离，同时这貌似和微前端理念不符合。
> 3. 如果我的系统暂时不存在react应用，目前只有 vue应用、jquery老系统，远程调用怎么做呢？

有iframe需求 就可以考虑用emp模式替换 换取更好的性能吧，毕竟iframe的消耗比较大特别是对于移动端来说。
